app/move/controllers/unassign.js

```diff
@@ -0,0 +1,60 @@
+const FormWizardController = require('../../../common/controllers/form-wizard')
+const allocationService = require('../../../common/services/allocation')
+const moveService = require('../../../common/services/move')
+
+class UnassignController extends FormWizardController {
+  middlewareChecks() {
+    this.use(this.checkAllocation)
+    super.middlewareChecks()
+  }
+
+  async checkAllocation(req, res, next) {
+    const { move } = res.locals
+
+    if (!move.allocation) {
+      return res.redirect(`/move/${move.id}`)
+    }
+    if (!move.person) {
+      return res.redirect(`/allocation/${move.allocation.id}`)
+    }
+
+    next()
+  }
+
+  middlewareLocals() {
+    super.middlewareLocals()
+    this.use(this.setMoveRelationships)
+  }
+
+  async setMoveRelationships(req, res, next) {
+    const { move } = res.locals
+
+    const allocation = await allocationService.getById(move.allocation.id)
```

Comment:
Just spotted you'll need a `try/catch` around this to catch any promise failures.

That also then led me to ask whether we need to make this call at all as `allocation` is available as a relationship on move: `move.allocation`?

Reply:
👍 


As for the necessity of the call, we do need it as we don't have the moves on the move's allocation relationship object. Maybe once we've properly implemented json:api includes we can return to this (and many other bits)

Reply:
^^ Doh! No, you're right, now that we're not passing the ids for move, allocation and person to double-check what we're removing, there is zero need  for the allocation call.

Zapped.